### PR TITLE
Android: Document Provider improvements

### PR DESCRIPTION
**Pull request by K0bin**
Fixes a bunch of bugs with the DocumentProvider and adds the ability to copy, delete and move folders.

I've also changed the app names of debug and release builds to avoid confusion for developers who have multiple versions of Dolphin (such as the Play Store release build + DEBUG) installed.

### CODE REVIEW

**Comment by t895** [comment]
@dolphin-emu-bot rebuild

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/DocumentProvider.kt [hunk 1]
```diff
@@ -111,67 +112,54 @@ class DocumentProvider : DocumentsProvider() {
...
+        rootDirectory ?: return null
+
         val folder = documentIdToPath(parentDocumentId)
         val file = findFileNameForNewFile(File(folder, displayName))
         if (mimeType == DocumentsContract.Document.MIME_TYPE_DIR) {
             file.mkdirs()
         } else {
             file.createNewFile()
         }
+        Log.w("DOCUMENTPROVIDER", file.absolutePath)
```
Comment (t895) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/DocumentProvider.kt:125]: This isn't particularly useful currently. Would you want to prefix it with "Created file <file>" or just get rid of it?

**Review by t895 — APPROVED** [review]
Thanks for reminding me to check this out. Just did a quick lookover and test. Thanks for the change and LGTM (besides that small comment).

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/DocumentProvider.kt (hunk 1 above)
Comment (K0bin) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/DocumentProvider.kt:125]: Oops, that's a debugging leftover.

Comment (K0bin) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/DocumentProvider.kt]: Removed.

**Review by lioncash — APPROVED** [review]

**Pull request merged**